aesara/tensor/basic.py

```diff
@@ -978,7 +978,9 @@ def zeros_like(model, dtype=None, opt=False):
 
 def zeros(shape, dtype=None):
     """Create a `TensorVariable` filled with zeros, closer to NumPy's syntax than ``alloc``."""
-    if not isinstance(shape, (np.ndarray, Sequence, TensorVariable)):
+    if (not isinstance(shape, (np.ndarray, Sequence, TensorVariable))) or (
+        isinstance(shape, TensorVariable) and shape.ndim == 0
+    ):
```

Comment:
I realize now this can be further simplified:

```suggestion
    if (not isinstance(shape, (np.ndarray, Sequence))) or (
        isinstance(shape, TensorVariable) and shape.ndim == 0
    ):
```

Reply:
No, remove `TensorVariable`  will lead to fail a test (I checked it before already):

`FAILED tests/tensor/test_basic.py::TestAlloc::test_zeros - ValueError: Tensor of type TensorType(float64, (None,)) could not be cast to have 0 dimensions`

Reply:
Ah so we want to test on the negative... 

```python
if not (
  isinstance(shape, (np.ndarray, Sequence)) 
  or (isinstance(shape, TensorVariable) and shape.ndim > 0)
):
```

Reply:
No, this will also fail.

Reply:
This one will work, so "in `tests` we trust" then :)
```
    if (not isinstance(shape, (np.ndarray, Sequence, TensorVariable))) or (
        isinstance(shape, TensorVariable) and shape.ndim == 0
    ):
```

Reply:
It's not super important, but there is a combination with one less isinstance check that must work, because there are no other possibilities besides `(not TensorVariable, TensorVariable.ndim == 0, TensorVariable.ndim > 0)`

Reply:
If `not TensorVariable` do: `shape = [shape]`
**OR**
If `is TensorVarialbe and ndim == 0` do: `shape == [shape]`

Can be converted to: 

If **not** `( is TensorVariable and ndim > 0)` do: `shape == [shape]`

The latter is shorter, but more difficult to understand :D

Reply:
Hi @ricardoV94, if you still prefer this option, I will change it to:

```
    if (not isinstance(shape, (np.ndarray, Sequence))) or (
        not (isinstance(shape, TensorVariable) and shape.ndim > 0)
    ):
```

Reply:
Hi @danhphan 

I tried this one I mentioned before locally and the tests seem to pass. What tests were failing?

> ```python
> if not (
>   isinstance(shape, (np.ndarray, Sequence)) 
>   or (isinstance(shape, TensorVariable) and shape.ndim > 0)
> ):
> ```

Reply:
Uhm, this will also pass locally at my end. 

I think the previous one, I put the `not` in wrong place like this:
```
if (
  not isinstance(shape, (np.ndarray, Sequence)) 
  or (isinstance(shape, TensorVariable) and shape.ndim > 0)
):
```
Which leads to:
`FAILED tests/tensor/test_basic.py::TestAlloc::test_zeros - ValueError: Tensor of type TensorType(float64, (None,)) could not be cast to have 0 dimensions`

So, I will then change it to:
```
 if not (
  isinstance(shape, (np.ndarray, Sequence)) 
  or (isinstance(shape, TensorVariable) and shape.ndim > 0)
):
```